### PR TITLE
feat: v2 migration Cloudflare KV

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Prepares for the [Verify Server V2 migration](https://github.com/WalletConnect/cloud-app/pull/1233) by mirroring all attestation writes to Cloudflare KV. Also reads attestations from Cloudflare KV if not found in Redis.

Remaining work:
- [x] What is the status code/response for a key not found?
- [x] Add CF API bearer token TF Cloud secret
- [ ] ~~Tests~~ Already covered by integration tests + tested manually
- [x] Merge https://github.com/WalletConnect/cloud-app/pull/1233
- [ ] Remove unnecessary CF API key & TF secrets

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/attestation_store/migration.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+use {
+    super::{cf_kv::CloudflareKv, AttestationStore, Result},
+    crate::util::redis,
+    async_trait::async_trait,
+};
+
+pub struct MigrationStore {
```
Comment (xDarksome) [inline src/attestation_store/migration.rs:7]: `migration::Store` is much nicer than `migration::MigrationStore`

Not exposing this type is even nicer if you don't need to, similar how `redis` returns `-> impl AttestationStore`

**Review by xDarksome — APPROVED** [review]
No need to fix any of this, just for the future

Liking the composability of the abstractions already? :smile:

**Comment by chris13524** [comment]
> Liking the composability of the abstractions already?

Actually they constrained what I wanted to implement: I wanted to call the regular get attestation endpoint from the worker, but this did an additional lookup on the data lake API which would've added latency and be redundant with the lookup that's being done here. If the abstraction wasn't so strong it probably would've been easier to use the response from the worker exclusively and disable the redundant lookup here.

Instead I made a new endpoint on the worker that didn't do the data lake call so that it would fit into the AttestationStore abstraction.

File: src/attestation_store/migration.rs (hunk 1 above)
Comment (chris13524) [reply src/attestation_store/migration.rs]: Fixed in 8e87e0ade08429121114eb31765e693324e8b2b5

File: src/attestation_store/cf_kv.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        let url = self
+            .endpoint
+            .join(&format!("/v1/compat-attestation/{id}"))?;
+        let response = self
+            .http_client
+            .get(url)
+            .timeout(Duration::from_secs(1))
+            .send()
+            .await?;
+        if response.status().is_success() {
```
[the remark this answers is not in the document]
Comment (chris13524) [reply src/attestation_store/cf_kv.rs]: Changed in a8a2c552bc707796af9a6e64c1e52017d3879321

**Pull request merged**